code/modules/clothing/suits/marine_armor.dm

```diff
@@ -288,51 +288,265 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
+		if(spaceacillin < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("spaceacillin",spaceacillin)
+		if(tricordrazine < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("tricordrazine",tricordrazine)
+		details +=("Significant blood toxicity detected. Chelating agents and curatives administered. <b>Dosage: Dylovene: [dylovene]U | Spaceacillin: [spaceacillin]U | Tricordrazine: [tricordrazine]U</b></br>")
+		B18_tox_cooldown = world.time + B18_CHEM_COOLDOWN
+		handle_chem_cooldown(B18_TOX_CODE)
+		dose_administered = TRUE
+
+	if(wearer.traumatic_shock > B18_automed_pain && !B18_pain_cooldown)
+		var/oxycodone = REAGENTS_OVERDOSE * 0.66 - (wearer.reagents.get_reagent_amount("oxycodone") + 0.5)
+		var/tramadol = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("tramadol") + 0.5)
+		if(oxycodone < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("oxycodone",oxycodone)
+		if(tramadol < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("tramadol",tramadol)
+		details +=("User pain at performance impeding levels. Painkillers administered. <b>Dosage: Oxycodone: [oxycodone]U | Tramadol: [tramadol]U</b></br>")
+		B18_pain_cooldown = world.time + B18_CHEM_COOLDOWN
+		handle_chem_cooldown(B18_PAIN_CODE)
+		dose_administered = TRUE
+
+	if(dose_administered)
+		playsound(src,'sound/items/hypospray.ogg', 25, 0, 1)
+		details +=("Estimated [B18_CHEM_COOLDOWN/600] minute replenishment time for each dosage.")
+		to_chat(wearer, "<span class='notice'>\icon [src] beeps:</br> [details.Join(" ")]</span>")
+
+/obj/item/clothing/suit/storage/marine/specialist/proc/handle_chem_cooldown(code = B18_BRUTE_CODE)
+	if(code)
+		spawn(B18_CHEM_COOLDOWN)
+			switch(code)
+				if(B18_BRUTE_CODE)
+					if(B18_brute_cooldown)
+						B18_brute_cooldown = null
+						to_chat(wearer,"<span class='notice'>[src] beeps: Trauma treatment reservoir replenished.</span>")
+				if(B18_BURN_CODE)
+					if(B18_burn_cooldown)
+						B18_burn_cooldown = null
+						to_chat(wearer,"<span class='notice'>[src] beeps: Burn treatment reservoir replenished.</span>")
+				if(B18_OXY_CODE)
+					if(B18_oxy_cooldown)
+						B18_oxy_cooldown = null
+						to_chat(wearer,"<span class='notice'>[src] beeps: Oxygenation treatment reservoir replenished.</span>")
+				if(B18_TOX_CODE)
+					if(B18_tox_cooldown)
+						B18_tox_cooldown = null
+						to_chat(wearer,"<span class='notice'>[src] beeps: Toxicity treatment reservoir replenished.</span>")
+				if(B18_PAIN_CODE)
+					if(B18_pain_cooldown)
+						B18_pain_cooldown = null
+						to_chat(wearer,"<span class='notice'>[src] beeps: Painkiller reservoir replenished.</span>")
+			playsound(src,'sound/effects/refill.ogg', 25, 0, 1)
+
+/obj/item/clothing/suit/storage/marine/specialist/verb/configure_automedic()
+	set name = "Configure Automedic"
+	set category = "B18 Armor"
+	set src in usr
 
-	if(usr.get_active_hand())
-		to_chat(usr, "Your active hand must be empty.")
+	if(!usr.canmove || usr.stat || usr.is_mob_restrained() || usr != wearer )
```

Comment:
Use `usr.is_mob_incapacitated()`, it's the all-in-one proc for the first three checks.